django/api/admin.py

```diff
@@ -116,3 +117,73 @@ def message_user(
 @admin.register(GoElectricRebate)
 class GoElectricRebateAdmin(admin.ModelAdmin):
     pass
+
+
+@admin.register(InitiatedGoElectricRebateApplication)
+class InitiatedGoElectricRebateApplicationAdmin(admin.ModelAdmin):
+    search_fields = ["drivers_licence", "id", "status"]
+    # disable bulk actions
+    actions = None
+    exclude = (
+        "sin",
+        "doc1",
+        "doc2",
+        "user",
+        "spouse_email",
+        "status",
+        "address",
+        "city",
+        "postal_code",
+        "application_type",
+        "doc1_tag",
+        "doc2_tag",
+        "consent_personal",
+        "consent_tax",
+    )
+    readonly_fields = (
+        "id",
+        "last_name",
+        "first_name",
+        "middle_names",
+        "email",
+        "user_is_bcsc",
+        "drivers_licence",
+        "date_of_birth",
+        "tax_year",
+        "is_legacy",
+    )
+
+    def get_queryset(self, request):
+        return GoElectricRebateApplication.objects.filter(
+            status=GoElectricRebateApplication.Status.HOUSEHOLD_INITIATED
+        )
+
+    def get_inlines(self, request, obj=None):
+        # TODO update this to use the proper enum later.
+        if obj and obj.application_type == "household":
+            return [HouseholdApplicationInline]
+        else:
+            return []
+
+    def has_delete_permission(self, request, obj=None):
+        return False
+
+    def response_change(self, request, obj):
+        ret = super().response_change(request, obj)
+        if "cancel_application" in request.POST:
+            obj.status = GoElectricRebateApplication.Status.CANCELLED
+            obj.save(update_fields=["status"])
+        return ret
+
+    def message_user(
+        self,
+        request,
+        message,
+        level=messages.INFO,
+        extra_tags="",
+        fail_silently=False,
+    ):
+        revised_level = level
+        if "reject_application" in request.POST:
```

Comment:
should be "cancel application"; also consider pulling now duplicated code out into a function